source/shared/cpp/ObjectModel/ColumnSet.cpp

```diff
@@ -47,11 +47,84 @@ std::shared_ptr<BaseCardElement> ColumnSetParser::Deserialize(ParseContext& cont
 }
 
 void ColumnSet::DeserializeChildren(ParseContext& context, const Json::Value& value)
-{
-    // Parse Columns
-    auto cardElements =
-        ParseUtil::GetElementCollectionOfSingleType<Column>(context, value, AdaptiveCardSchemaKey::Columns, Column::Deserialize, false);
-    m_columns = std::move(cardElements);
+{ 
+    auto elementArray = ParseUtil::GetArray(value, AdaptiveCardSchemaKey::Columns, false);
+
+    std::vector<std::shared_ptr<Column>> elements;
+    if (elementArray.empty())
+    {
+        m_columns = elements;
+        return;
+    }
+
+    const size_t elemSize = elementArray.size();
+    const size_t endElemIndex = elemSize - 1;
+    elements.reserve(elemSize);
+
+    size_t currentIndex = 0;
+
+    // Child's bleed get more restricted as we walk down the tree
+    // until padding is reset
+    // padding get rested when padding is added to a collection type element
```

Comment:
>get rested [](start = 15, length = 10)

typo: `gets reset`?

Reply:
:-) thanks

---
In reply to: [268274205](https://github.com/Microsoft/AdaptiveCards/pull/2551#discussion_r268274205) [](ancestors = 268274205)